Plex: implement recommendations with configurable hub limit

anatosun opened the pull request:
This PR adds the recommendations (hubs) in the music assistant dashboard with a settable limit. 

I had some performance issues while testing this feature. It'd be great if someone else would test it locally.

anatosun commented on music_assistant/providers/plex/__init__.py:
```diff
@@ -1085,6 +1101,83 @@ async def get_similar_tracks(self, prov_track_id: str, limit: int = 25) -> list[
+                )
+
+                # Parse each item based on its type (limit to configured max)
+                for item in hub.items()[:limit]:
```
The item function does not take any argument unfortunately. If someone knows how to limit the items returned with the API, I'd be grateful.

marcelveldt replied:
Keep it like this for now (just make sure to use caching whenever possible) and optimiz elater if you see a more efficient way to do so

marcelveldt reviewed (approved):
Nice work, @anatosun !

OzGav commented:
@anatosun great to have you aboard. I look after the docs so if you keep those in mind when making an changes and ping me if you think something needs doing. I try and keep an eye on all the PRs but I don't want to miss any.

marcelveldt commented:
Small merge conflict to be resolved...

marcelveldt commented:
CI found an issue - looks like a small typo ?

anatosun commented:
Converted to draft, since I realized a small mistake in the code; will fix in the coming days!

anatosun committed:
Plex: implement recommendations with configurable hub limit

Add support for Plex recommendations via hubs with a configurable
limit for items per hub. The implementation fetches music hubs from
the Plex library and parses tracks, albums, artists, and playlists.

- Add CONF_HUB_ITEMS_LIMIT config entry (default: 10, range: 1-100)
- Implement recommendations() method using Plex library hubs API
- Properly wrap synchronous Plex API calls with _run_async()
- Parse hub items into RecommendationFolder objects
- Add ProviderFeature.RECOMMENDATIONS to supported features

anatosun committed:
Plex: fix recommendations performance and hub fetching

anatosun commented:
I made a few optimizations to improve performance
